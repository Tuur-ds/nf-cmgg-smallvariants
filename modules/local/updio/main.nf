process UPDIO {
    tag "$meta.id"
    label 'process_single'

    container "cmgg/updio:1.0.0"

    input:
    tuple val(meta), path(vcf), path(tbi)
    tuple val(meta2), path(cnv)

    output:
    tuple val(meta), path("${prefix}"), emit: updio
    path  "versions.yml"              , emit: versions

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"

    def common_cnv_file = cnv ? "--common_cnv_file $cnv" : "--common_cnv_file /usr/local/lib/updio/sample_data/common_dels_1percent_liftover.tsv"
    def VERSION = "1.0.0"

    """
    UPDio \\
        --multisample_vcf $vcf \\
        --output_path $prefix \\
        $common_cnv_file \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        UPDio: $VERSION
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"

    def VERSION = "1.0.0"
    """
    mkdir $prefix
    touch ${prefix}/${meta.child}.events_list
    touch ${prefix}/${meta.child}.log
    touch ${prefix}/${meta.child}.table
    touch ${prefix}/${meta.child}.upd

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        UPDio: $VERSION
    END_VERSIONS
    """
}
