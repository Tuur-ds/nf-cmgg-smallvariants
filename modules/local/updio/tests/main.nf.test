nextflow_process {

    name "Test Process UPDIO"
    script "../main.nf"
    process "UPDIO"

    tag "modules"
    tag "modules_local"
    tag "updio"

    test("homo_sapiens - vcf, tbi, []") {

        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', child:'NA24385' ], // meta map
                    file(params.famvcf, checkIfExists:true),
                    file(params.famtbi, checkIfExists:true)
                ]
                input[1] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - vcf, tbi, [] - stub") {

        options "-stub"
        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', child:'NA24385' ], // meta map
                    file(params.famvcf, checkIfExists:true),
                    file(params.famtbi, checkIfExists:true)
                ]
                input[1] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}