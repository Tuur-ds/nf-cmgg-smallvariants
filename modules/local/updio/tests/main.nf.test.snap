{
    "homo_sapiens - vcf, tbi, [] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "child": "NA24385"
                        },
                        [
                            "NA24385.events_list:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "NA24385.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "NA24385.table:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "NA24385.upd:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,8a807f51c31f64de1c707210fabe7029"
                ],
                "updio": [
                    [
                        {
                            "id": "test",
                            "child": "NA24385"
                        },
                        [
                            "NA24385.events_list:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "NA24385.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "NA24385.table:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "NA24385.upd:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8a807f51c31f64de1c707210fabe7029"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-14T09:57:26.032830013"
    },
    "homo_sapiens - vcf, tbi, []": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "child": "NA24385"
                        },
                        [
                            "NA24385.events_list:md5,4a26c133ef193d31eddfd18dae94f0a0",
                            "NA24385.log:md5,a40a24f379127a9cde7e40a1ce1032ec",
                            "NA24385.table:md5,ca8165fc7869a113ca034396de7cf579",
                            "NA24385.upd:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,8a807f51c31f64de1c707210fabe7029"
                ],
                "updio": [
                    [
                        {
                            "id": "test",
                            "child": "NA24385"
                        },
                        [
                            "NA24385.events_list:md5,4a26c133ef193d31eddfd18dae94f0a0",
                            "NA24385.log:md5,a40a24f379127a9cde7e40a1ce1032ec",
                            "NA24385.table:md5,ca8165fc7869a113ca034396de7cf579",
                            "NA24385.upd:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8a807f51c31f64de1c707210fabe7029"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-14T09:57:17.522197249"
    }
}